core/src/consensus/validator_set/dynamic_validator.rs

```diff
@@ -143,6 +143,23 @@ impl ValidatorSet for DynamicValidator {
         }
     }
 
+    fn normalized_voting_power(&self, parent: &BlockHash, index: usize, total_power: u64) -> Result<u64, EngineError> {
+        if let Some(validators) = self.validators(*parent) {
+            let validator = validators.get(index).ok_or_else(|| {
+                EngineError::ValidatorNotExist {
+                    height: 0, // FIXME
```

Comment:
I fixed it by adding a parameter named "height". Because `DynamicValidator` holds a `client`, the client look up block numbers by block hashes. However, It is possible there is no matched blockNumber to the input block hash. I don't want such handling in the function. So I chose to inject height information from the caller of the function.